app/helpers/application_helper.rb

```diff
@@ -142,10 +142,21 @@ def preload_script(script)
   end
 
   def preload_script_url(url)
-    <<~HTML.html_safe
-      <link rel="preload" href="#{url}" as="script">
-      <script defer src="#{url}"></script>
-    HTML
+    add_resource_preload_list(url, 'script')
+    if ENV['PRELOAD_LINK_HEADER']
```

Comment:
Is there a reason for using an ENV directly instead of `GlobalSetting.preload_link_header`? (which will be populated by the `DISCOURSE_PRELOAD_LINK_HEADER` env)

Functionality is the same, but global setting approach is more common in our codebase I think?

Reply:
See last commit